icechunk/src/config.rs

```diff
@@ -0,0 +1,93 @@
+use serde::{Deserialize, Serialize};
+
+use crate::virtual_chunks::{
+    mk_default_containers, ContainerName, VirtualChunkContainer,
+};
+
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq)]
+pub struct S3CompatibleOptions {
+    pub region: Option<String>,
+    pub endpoint_url: Option<String>,
+    pub anonymous: bool,
+    pub allow_http: bool,
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq)]
+pub enum ObjectStoreConfig {
```

Comment:
The only thing I don't like is that this isn't extensible..for instance, for wasm we may need a different storage provider only for wasm, this means for every storage implementation we need to update this enum, there's no way for third parties to add their own storage implementations

Reply:
I agree. I feel this is high level API that should be overridable. We can add a more powerful mechanism later.